### PR TITLE
Initial commit of third JAX-101 notebook

### PR DESCRIPTION
**Pull request by jakevdp**
To be merged after #5905

Still TODO: authors

### CODE REVIEW

File: docs/jax-101/03-vectorization.ipynb [hunk 1]
```diff
@@ -0,0 +1,376 @@
...
+    "id": "zMIrmiaZxiJC"
+   },
+   "source": [
+    "# Automatic vectorization in JAX\n",
+    "\n",
+    "[![Open in Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/google/jax/blob/master/docs/jax-101/03-vectorization.ipynb)\n",
+    "\n",
+    "*Authors: TODO*\n",
+    "\n",
+    "In the previous section we discussed JIT compilation via the `jax.jit` function. This notebook discusses another of JAX's transforms: vectorization via `jax.vmap`."
```
Comment (8bitmp3) [inline docs/jax-101/03-vectorization.ipynb:15]: Suggestion: provide a link to the "previous section" for better UX (as in "[previous section](....awesome link....)")

File: docs/jax-101/03-vectorization.ipynb [hunk 2]
```diff
@@ -0,0 +1,376 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "bsxT4hA6RTCG"
+   },
+   "source": [
+    "## Combining transformations\n",
+    "\n",
+    "As with all JAX transformations, `jax.jit` and `jax.vmap` are designed to be composable, which means you can wrap a vmapped function with JIT, or a JITted function with vmap, and everything will work correctly:"
```
Comment (8bitmp3) [inline docs/jax-101/03-vectorization.ipynb:316]: Suggestion:

```suggestion
    "As with all JAX transformations, `jax.jit` and `jax.vmap` are designed to be composable, which means you can wrap a vmapped function with JIT, or a JITted function with `vmap`, and everything will work correctly:"
```

File: docs/jax-101/03-vectorization.ipynb [hunk 3]
```diff
@@ -0,0 +1,376 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "VTh0l_1SUlh4"
+   },
+   "source": [
+    "This produces the correct result, however it is not very efficient.\n",
+    "\n",
+    "In order to batch the computation efficiently, you would normally have to rewrite the function manually to ensure it is done in vectorised form. This is not particularly difficult to implement, but does involve changing how the function treats indices, axes, and other parts of the input.\n",
```
Comment (8bitmp3) [inline docs/jax-101/03-vectorization.ipynb:143]: Suggestion: nit 🇬🇧 vs 🇺🇸 "vectorized" (I prefer "vectorised" 😎 though but the tech style guides recommend 🇺🇸 English 🤷 )

```suggestion
    "In order to batch the computation efficiently, you would normally have to rewrite the function manually to ensure it is done in vectorized form. This is not particularly difficult to implement, but does involve changing how the function treats indices, axes, and other parts of the input.\n",
```

File: docs/jax-101/03-vectorization.ipynb [hunk 4]
```diff
@@ -0,0 +1,376 @@
...
+    "auto_batch_convolve(xs, ws)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "7aVAy7332lFj"
+   },
+   "source": [
+    "It does this by tracing the function similarly to `jax.jit`, and automatically adding batch axes at the beggining of each input.\n",
```
Comment (8bitmp3) [inline docs/jax-101/03-vectorization.ipynb:234]: Suggestion: tiny typo ("beginning")

```suggestion
    "It does this by tracing the function similarly to `jax.jit`, and automatically adding batch axes at the beginning of each input.\n",
```

File: docs/jax-101/03-vectorization.ipynb (hunk 1 above)
Comment (jakevdp) [reply docs/jax-101/03-vectorization.ipynb:15]: Same issue as in notebook 2... we don't know if people are reading this as a notebook on github or as a page in the docs, so it's not clear which link to include.

Comment (8bitmp3) [reply docs/jax-101/03-vectorization.ipynb:15]: Gotcha @jakevdp thank you

**Review by skye — APPROVED** [review]

**Pull request merged**